### DISCUSSION TITLE
DataTemplate Selectors in Collection views

### DISCUSSION THREAD

**Discussion by roughiain** [CommunityToolkit/Maui.Markup#180] [Q&A] [answered]
Looking at the hacker news sample and was looking how to implement a DataTemplateSelector and could not see any obvious ways do it. Normal XAML you would have the template selector as a Resource in the page then use {StaticResource x:Key} to bind it. I could be missing something very obvious. Any pointers would be appreciated.

**Comment by TheCodeTraveler** [comment, marked as answer]
Hi @roughiain! 

Are you looking for help creating a `DataTemplateSelector` using C# Markup? Here's an example of how to do it from my app, GitTrends: https://github.com/brminnick/GitTrends/blob/main/GitTrends/Views/Repository/RepositoryDataTemplateSelector.cs

Or are you asking how to add a `DataTemplateSelector` to a `CollectionView` using the C# Markup Extensions?  You do it the same way you would do it for a `DataTemplate`:

```cs
new CollectionView()
  .ItemTemplate(new MyDataTemplateSelector());
```

**Reply by roughiain** [reply]
Thanks @brminnick was missing the obvious

**Comment by CliffAgius** [comment]
Closed and answered
